packages/ng-primitives/date-picker/src/date-picker-previous-month/date-picker-previous-month.spec.ts

```diff
@@ -0,0 +1,189 @@
+import { BooleanInput } from '@angular/cdk/coercion';
+import {
+  Component,
+  InputSignal,
+  InputSignalWithTransform,
+  Provider,
+  signal,
+  WritableSignal,
+  viewChild,
+} from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { NgpNativeDateAdapter } from 'ng-primitives/date-time';
+import { NgpDatePicker } from '../date-picker/date-picker';
+import { NgpDatePickerStateToken } from '../date-picker/date-picker-state';
+import { NgpDatePickerPreviousMonth } from './date-picker-previous-month';
+
+describe('NgpDatePickerPreviousMonth', () => {
+  let fixture: ComponentFixture<TestHost>;
+  let host: TestHost;
+  let focusedDateSignal: WritableSignal<Date>;
+  let disabledSignal: WritableSignal<boolean>;
+  let minSignal: WritableSignal<Date | undefined>;
+  let dateAdapter: NgpNativeDateAdapter;
+
+  beforeEach(() => {
+    dateAdapter = new NgpNativeDateAdapter();
+    focusedDateSignal = signal(new Date());
+    disabledSignal = signal(false);
+    minSignal = signal(undefined);
+
+    TestBed.configureTestingModule({
+      imports: [TestHost],
+      providers: [provideMockDatePickerState(focusedDateSignal, disabledSignal, minSignal)],
+    });
+    fixture = TestBed.createComponent(TestHost);
+    fixture.detectChanges();
+    host = fixture.componentInstance;
+  });
+
+  it('should create', () => {
+    expect(host.previousMonth()).toBeTruthy();
+  });
+
+  it('should navigate to previous month when clicked', () => {
+    const currentDate = new Date(2025, 2, 15); // March 15, 2025
+    focusedDateSignal.set(currentDate);
+
+    const button = fixture.nativeElement.querySelector('button');
+    button.click();
+    fixture.detectChanges();
+
+    const newDate = focusedDateSignal();
+    expect(dateAdapter.getMonth(newDate)).toBe(1); // February
+    expect(dateAdapter.getYear(newDate)).toBe(2025);
+    expect(dateAdapter.getDate(newDate)).toBe(1); // First day of the month
+  });
+
+  it('should handle navigation from 31st of month to previous month without 31 days', () => {
```

Comment:
I went ahead and added tests for this file since they didn't exists, but this was the specific test case that covers the issue being addressed in this PR.